src/diffusers/models/transformers/cogvideox_transformer_3d.py

```diff
@@ -452,7 +452,7 @@ def forward(
 
         # 3. Transformer blocks
         for i, block in enumerate(self.transformer_blocks):
-            if self.training and self.gradient_checkpointing:
+            if self.gradient_checkpointing:
```

Comment:
oh thanks! why do we also removed the `torch.is_grad_enabled()` check? gradient checkpointing isn't meaningful without gradient being computed, no?

Reply:
added it back, thanks for pointing it out. 
it does not break anything, but found that it throws an annoying warning when use_reentrant=True,

Reply:
>but found that it throws an annoying warning when use_reentrant=True,

what do you mean by that?

Reply:
use_reentrant is an argument passed to torch.utils.checkpoint.checkpoint

if True one of the checks will print this to stderr
warnings.warn(
            "None of the inputs have requires_grad=True. Gradients will be None"
        )
 but diffusers are using use_reentrant=False anyway

Reply:
oh got thanks, so the warning is specific to when we use gradient checkpointing when gradient is not enabled